great_expectations/data_context/data_context.py

```diff
@@ -405,6 +406,25 @@ def save_config_variable(self, config_variable_name, value):
         with open(config_variables_filepath, "w") as config_variables_file:
             yaml.dump(config_variables, config_variables_file)
 
+    def delete_datasource(self,datasource_name=None):
+        """Delete data source 
+        Args:
+
+        Returns:
+        """
+        if datasource_names is None:
+            raise ValueError(
+                "Datasource names must be a datasource name"
+            )
+        else:
+            datasource = self.get_datasource(datasource_name)
+            if datasource:
+               self._datasources.remove(datasource_name)
```

Comment:
I believe instead of removing the in-memory datasource, this should remove the datasource from the configuration.

Reply:
WIP at the time, looks updated

Reply:
WIP at the time resolved since